tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -603,26 +605,37 @@ class AdaptiveQueryExecSuite
   test("SPARK-35585: Support propagate empty relation through project/filter") {
     logError("SPARK-35585: Support propagate empty relation through project/filter")
     assumeSpark320orLater
-
+    // There is difference in Spark plans between 3.2.x-3.5.x and 4.0+
+    // Spark 4.0+ has EmptyRelationExec, but 3.2.x-3.5.x has LocalTableScanExec as the
+    // root node of the plan.
+    // Issue to support EmptyRelationExec:https://github.com/NVIDIA/spark-rapids/issues/11100
     val conf = new SparkConf()
         .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
         .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, "-1")
         .set(RapidsConf.TEST_ALLOWED_NONGPU.key,
-          "DataWritingCommandExec,ShuffleExchangeExec,HashPartitioning")
+          "DataWritingCommandExec,ShuffleExchangeExec,HashPartitioning,EmptyRelationExec")
 
     withGpuSparkSession(spark => {
       testData(spark)
 
       val (plan1, adaptivePlan1) = runAdaptiveAndVerifyResult(spark,
         "SELECT key FROM testData WHERE key = 0 ORDER BY key, value")
       assert(findTopLevelSort(plan1).size == 1)
-      assert(stripAQEPlan(adaptivePlan1).isInstanceOf[LocalTableScanExec])
+      // Check for either LocalTableScanExec (pre-4.0) or EmptyRelationExec (4.0+)
+      val strippedPlan1 = stripAQEPlan(adaptivePlan1)
+      assert(strippedPlan1.isInstanceOf[LocalTableScanExec] ||
+            strippedPlan1.getClass.getSimpleName.equals("EmptyRelationExec"),
```

Comment:
ideally it should be properly shimmed as opposed to string match with getSimpleName. Otherwise it introduces inconsistency with the LocalTableScanExec check.